x/autopilot/keeper/liquidstake.go

```diff
@@ -72,12 +72,47 @@ func (k Keeper) RunLiquidStake(ctx sdk.Context, addr sdk.AccAddress, token sdk.C
 	}
 
 	msgServer := stakeibckeeper.NewMsgServerImpl(k.stakeibcKeeper)
-	_, err := msgServer.LiquidStake(
+	result, err := msgServer.LiquidStake(
 		sdk.WrapSDKContext(ctx),
 		msg,
 	)
 	if err != nil {
-		return errorsmod.Wrapf(sdkerrors.ErrInsufficientFunds, err.Error())
+		return errorsmod.Wrapf(err, err.Error())
 	}
-	return nil
+
+	if packetMetadata.IbcReceiver == "" {
+		return nil
+	}
+
+	hostZone, err := k.stakeibcKeeper.GetHostZoneFromHostDenom(ctx, token.Denom)
+	if err != nil {
+		return err
+	}
+
+	return k.IBCTransferStAsset(ctx, result.StToken, addr.String(), hostZone, packetMetadata)
+}
+
+func (k Keeper) IBCTransferStAsset(ctx sdk.Context, stAsset sdk.Coin, sender string, hostZone *stakeibctypes.HostZone, packetMetadata types.StakeibcPacketMetadata) error {
+	ibcTransferTimeoutNanos := k.stakeibcKeeper.GetParam(ctx, stakeibctypes.KeyIBCTransferTimeoutNanos)
+	timeoutTimestamp := uint64(ctx.BlockTime().UnixNano()) + ibcTransferTimeoutNanos
+	channelId := packetMetadata.TransferChannel
+	if channelId == "" {
+		channelId = hostZone.TransferChannelId
+	}
+	transferMsg := &transfertypes.MsgTransfer{
+		SourcePort:    transfertypes.PortID,
+		SourceChannel: channelId,
+		Token:         stAsset,
+		// TODO: does this reintroduce the bug in PFM where senders can be spoofed?
+		// If so, should we instead call PFM directly to forward the packet?
+		// Or should we obfuscate the sender, making it a random address?
+		Sender:           sender,
```

Comment:
Based on our discussion yesterday, I think we should make a point to just not trust senders

Can you give some more details around these two solutions? Specifically:
* how complex is it to do the obfuscation ourselves?
* would using PFM avoid the need for us to implement our own callback?
* what do you consider the tradeoffs between these? my gut reaction is that using PFM might simplify the implementation but make things a bit harder to reason about

Reply:
The only potential worry I have about obfuscating the sender is what happens if the IBC transfer fails, does it get returned to the obfuscated sender?

Reply:
we discussed this on a call - decided to move forward with obfuscating the sender and using a fallback address. marking this resolved (since it will be implemented in a separate PR)